homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,146 @@
+"""Home Assistant Switcher Component.
+
+For controlling the Switcher Boiler Device (https://www.switcher.co.il/).
+Please follow configuring instructions here:
+    https://www.home-assistant.io/components/switcher_kis/
+
+Author: Tomer Figenblat
+
+Example yaml configuration,
+Get you device details using the instructions here:
+https://github.com/NightRang3r/Switcher-V2-Python.
+
+Mandatory only configuration:
+switcher_kis:
+  phone_id: 'xxxx'
+  device_id: 'xxxxxx'
+  device_password: 'xxxxxxxx'
+
+Full configuration:
+switcher_kis:
+  phone_id: 'xxxx'
+  device_id: 'xxxxxx'
+  device_password: 'xxxxxxxx'
+  name: 'my_boiler'
+  friendly_name: "My Friendly Boiler"
+  icon: 'mdi:my-icon'
+  include_schedule_sensors: true
+  schedules_scan_interval:
+    minutes: 5
+
+Defaults:
+name = 'boiler'
+friendly_name = 'Boiler'
+include_sensors = false
+schedules_scan_interval = {'mintues': 5}
+
+The minimal configuration will:
+- Create 1 switch entity called Boiler with the entity id:
+  switch.switcher_kis_boiler.
+- Register the following services:
+  - switcher_kis.turn_on - no arguments needed.
+  - switcher_kis.turn_off - no arguments needed.
+  - switcher_kis.turn_on_15_minutes - no arguments needed.
+  - switcher_kis.turn_on_30_minutes - no arguments needed.
+  - switcher_kis.turn_on_45_minutes - no arguments needed.
+  - switcher_kis.turn_on_60_minutes - no arguments needed.
+  - switcher_kis.set_auto_off - takes 1 string argument {'auto_off': '01:30'}
+  - switcher_kis.update_device_name - takes 1 string argument {'name': 'boil'}
+
+Setting the 'include_schedule_sensors' key to True,
+Will result in all the above plus the following:
+- Creation of 8 sensors entities represnting the device's schedules (0-7),
+  The names will be constructed from the 'name' key '_schedule' and the id,
+  The same goes for the friendly names:
+  - sensor.switcher_kis_boiler_schedule0: Boiler Schedule 0
+  ...
+  - sensor.switcher_kis_boiler_schedule7: Boiler Schedule 7
+
+  The sensors state will allways provide the minumum information:
+  - Not configured
+  - Not enabled
+  - Due today at 17:30
+  - Due tommorow at 20:00
+
+- Register the following services:
+  - switcher_kis.create_schedule - takes the following mandatory arguments:
+    {'start_time': '19:30', 'end_time': '20:30', 'recurring': false}
+    if recurring is set to true, you must provide an extra argument:
+    {'days': ['Monday', 'Wednesday', 'Saturday']}
+  - switcher_kis.delete_schedule - takes 1 int argument {'schedule_id': 3'}
+  - switcher_kis.enable_schedule - takes 1 int argument {'schedule_id': 0'}
+  - switcher_kis.disable_schedule - takes 1 int argument {'schedule_id': 7'}
+
+"""
```

Comment:
While you are totally right, we need to avoid dual-maintenance.
Please note that this part was added temporarily to allow code reviewers to review the code before the `home-assistant.io` docs was written.
Please note that in the [Development Checklist](https://developers.home-assistant.io/docs/en/development_checklist.html), it specifically  states:

> It's OK to start with adding a docstring with configuration details (for example, sample entry for configuration.yaml file) to the file header.

I was actually planning on dropping this verbose header once the reviews are done.
I will do so now.
:-)